bot/exts/evergreen/wikipedia.py

```diff
@@ -1,114 +1,99 @@
-        """Return list of results containing your search query from wikipedia."""
-        titles = await self.search_wikipedia(search)
-
-        def check(message: Message) -> bool:
-            return message.author.id == ctx.author.id and message.channel == ctx.channel
-
-        if not titles:
-            await ctx.send("Sorry, we could not find a wikipedia article using that search term")
-            return
-
-        async with ctx.typing():
-            log.info("Finished appending titles to titles_no_underscore list")
-
-            s_desc = "\n".join(self.formatted_wiki_url(index, title) for index, title in enumerate(titles, start=1))
-            embed = Embed(colour=Color.blue(), title=f"Wikipedia results for `{search}`", description=s_desc)
-            embed.timestamp = datetime.datetime.utcnow()
-            await ctx.send(embed=embed)
-        embed = Embed(colour=Color.green(), description="Enter number to choose")
-        msg = await ctx.send(embed=embed)
-        titles_len = len(titles)  # getting length of list
-
-        for retry_count in range(1, Wikipedia.total_chance + 1):
-            retries_left = Wikipedia.total_chance - retry_count
-            if retry_count < Wikipedia.total_chance:
-                error_msg = f"You have `{retries_left}/{Wikipedia.total_chance}` chances left"
-            else:
-                error_msg = 'Please try again by using `.wiki` command'
-            try:
-                message = await ctx.bot.wait_for('message', timeout=60.0, check=check)
-                response_from_user = await self.bot.get_context(message)
-
-                if response_from_user.command:
-                    return
-
-                response = int(message.content)
-                if response < 0:
-                    await ctx.send(f"Sorry, but you can't give negative index, {error_msg}")
-                elif response == 0:
-                    await ctx.send(f"Sorry, please give an integer between `1` to `{titles_len}`, {error_msg}")
-                else:
-                    await ctx.send(WIKIPEDIA_URL.format(title=titles[response - 1].replace(" ", "_")))
-                    break
-
-            except asyncio.TimeoutError:
-                embed = Embed(colour=Color.red(), description=f"Time's up {ctx.author.mention}")
-                await msg.edit(embed=embed)
-                break
-
-            except ValueError:
-                await ctx.send(f"Sorry, but you cannot do that, I will only accept an positive integer, {error_msg}")
-
-            except IndexError:
-                await ctx.send(f"Sorry, please give an integer between `1` to `{titles_len}`, {error_msg}")
-
-            except Exception as e:
-                log.info(f"Caught exception {e}, breaking out of retry loop")
-                break
-
-
-def setup(bot: commands.Bot) -> None:
+        """Sends paginated top 10 results of Wikipedia search.."""
+        contents = await self.wiki_request(ctx, search)
```

Comment:
You can just send the `channel` instead of sending the entire context with `ctx.channel`